### PR TITLE
Improve duration parsing

### PR DESCRIPTION
**Pull request by DenDen039**
##### SUMMARY
Improve duration parsing. Parser accepts combination of time units

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
fusion_pp

##### ADDITIONAL INFORMATION
Testing: unit-tests

### CODE REVIEW

**Review by sdodsley — DISMISSED** [review]
I like this new function. Very cool

File: plugins/module_utils/parsing.py [hunk 1]
```diff
@@ -4,11 +4,24 @@
...
 
 from __future__ import absolute_import, division, print_function
+import re
 
 __metaclass__ = type
 
 METRIC_SUFFIXES = ["K", "M", "G", "T", "P"]
 
+duration_pattern = re.compile(
+    r"^((?P<Y>[1-9]\d*)Y)?((?P<W>[1-9]\d*)W)?((?P<D>[1-9]\d*)D)?(((?P<H>[1-9]\d*)H)?((?P<M>[1-9]\d*)M)?)?$"
```
Comment (andreepyro) [inline plugins/module_utils/parsing.py:14]: This is impossible to read and hard to update in future. I suggest using some already existing libraries like `datetime` (e.g. function `datetime.strptime`).

**Review by dturecekps — DISMISSED** [review]
LGTM

**Review by andreepyro — CHANGES_REQUESTED** [review]
As formatting isn't anything new, I would use already existing solution to avoid bugs.

Kudos for adding the unit tests!

**Review by dturecekps — APPROVED** [review]

File: plugins/module_utils/parsing.py (hunk 1 above)
Comment (DenDen039) [reply plugins/module_utils/parsing.py:14]: It doesn't work in our case. `datetime.strptime` requires exact format of date. But we can have different combinations for duration. (e.g 10H5D, 3W5D, etc)

**Review by andreepyro — APPROVED** [review]

**Review by jankodera-ps — APPROVED** [review]

**Pull request merged**